### PR TITLE
boards: add Pinnacle 100 DVK

### PR DESCRIPTION
**Pull request by rerickson1**
Add Laird Connectivity Pinnacle 100 modem development kit.

This depends on https://github.com/zephyrproject-rtos/zephyr/pull/27676

### CODE REVIEW

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk_defconfig
Comment (carlescufi) [inline boards/arm/pinnacle_100_dvk/pinnacle_100_dvk_defconfig:34]: Are you sure you want networking enabled by default? You could instead leave it to the user

Comment (rerickson1) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk_defconfig:34]: We feel the point of using our modem is to use it for networking, thus enabling it by default to make it easier for the user out of the gate.

**Review by carlescufi — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]
Looks fine to me.
Guess it is weird how this was passing CI with .yaml file having a wrong name.
Also, it is the first time I see slot1_partition being defined on a separate memory container :)

**Comment by carlescufi** [comment]
@rerickson1 please take a look at the CI failures

**Comment by rerickson1** [comment]
> @rerickson1 please take a look at the CI failures

I believe all the CI errors relate to the HL7800 driver not being present yet.  Will rebase this once #27676 is approved and merged.

File: boards/arm/bl654_dvk/doc/bl654_dvk.rst
Comment (MaureenHelm) [inline boards/arm/bl654_dvk/doc/bl654_dvk.rst:1]: Did you mean to remove this file?

File: boards/arm/pinnacle_100_dvk/doc/index.rst
Comment (MaureenHelm) [inline boards/arm/pinnacle_100_dvk/doc/index.rst:138]: This seems to be in conflict the debugging section below?

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.yaml
Comment (MaureenHelm) [inline boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.yaml:2]: Did you mean to move this file? That will leave the Adafruit board without a yaml file

File: drivers/modem/hl7800.c
Comment (MaureenHelm) [inline drivers/modem/hl7800.c:4629]: The struct device changes should be in a separate commit from the newlibc changes. Both need a better commit message explanation.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/bl654_dvk/doc/bl654_dvk.rst
Comment (rerickson1) [reply boards/arm/bl654_dvk/doc/bl654_dvk.rst]: Rebase error, thanks for catching this.

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.yaml
Comment (rerickson1) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.yaml]: Rebase error

File: samples/philosophers/sample.yaml
Comment (MaureenHelm) [inline samples/philosophers/sample.yaml:33]: why are you explicitly excluding this platform for so many samples? will you please explain in your commit message?

**Review by MaureenHelm — COMMENTED** [review]
thanks for splitting up the struct device and newlibc changes. just one more question

File: samples/philosophers/sample.yaml
Comment (rerickson1) [reply samples/philosophers/sample.yaml:33]: A reason for some of these is that I could not add a board kcofnig overlay and get it to work. I believe there is an issue.  I filed it here https://github.com/zephyrproject-rtos/zephyr/issues/27934.

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.yaml
Comment (jfischer-phytec-iot) [inline boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.yaml:23]: If "Pinnacle 100 (HL7800 driver) does not support IPv6" then why is here "netif:openthread" present?

**Review by MaureenHelm — APPROVED** [review]

File: tests/kernel/sched/schedule_api/testcase.yaml
Comment (nashif) [inline tests/kernel/sched/schedule_api/testcase.yaml:13]: why is this being excluded? You need to add a comment about this.

File: tests/drivers/console/testcase.yaml
Comment (nashif) [inline tests/drivers/console/testcase.yaml:17]: why is this being excluded? You need to add a comment about this.

**Review by nashif — CHANGES_REQUESTED** [review]
Why are you excluding all of those samples and tests, there is something not right here and no explanation of why those are being excluded.

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk_defconfig
Comment (jfischer-phytec-iot) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk_defconfig:34]: I will block this PR because of `CONFIG_NETWORKING=y` and `CONFIG_NET_IPV6=n`, which leads to unnecessary changes in tests and samples. This board is based on nRF52840 and therefore also supports Bluetooth LE, 802154, and IPv6. If nRF52840 does not have any RF functionality, then the items "ble, ieee802154" must be removed. 

And in general it is not necessary to enable CONFIG_NETWORKING on board level. Which networking sample does not do it anyway?

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Comment by lairdjm** [comment]
@jfischer-phytec-iot the issue was that the HL7800 driver doesn't support IPv6 and (in Zephyr 1.14 at least, hasn't been tested since) if any sample apps which had IPv6 sockets (e.g. http) in them were compiled and ran, they would instantly hardfault, the IPv6 Bluetooth radio was never tested. The board has networking enabled by default because it is a cellular modem module, it wouldn't make sense to use this particular board without networking.

**Review by jfischer-no — APPROVED** [review]

**Comment by ioannisg** [comment]
@rerickson1 moving this to 2.5.0 release (you can add TSC-label to request handling this exceptionally and include it in 2.4.0)

**Comment by rerickson1** [comment]
> @rerickson1 moving this to 2.5.0 release (you can add TSC-label to request handling this exceptionally and include it in 2.4.0)

Moving back to 2.4.0 and requesting TSC approval. This PR was started before the merge window closed. This PR fixes a few issues that prevent the HL7800 driver from compiling. Also, this should be very low risk to get in 2.4.0. @nashif can I get you to re-review?

**Comment by jfischer-phytec-iot** [comment]
> This PR fixes a few issues that prevent the HL7800 driver from compiling.

You could move the fixes to a new PR and not to take this PR as "hostage".

**Review by nashif — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by rerickson1** [comment]
Approved for inclusion of v2.4.0 based on TSC vote on 9-16-2020.

**Pull request merged**
